_posts/en/newsletters/2023-10-25-newsletter.md

```diff
@@ -0,0 +1,588 @@
+        replacement cycle.
+
+    - **OP_EXPIRE:** in a separate thread, but quoting from the
+      replacement cycles thread, Peter Todd [proposed][todd expire1]
+      several consensus changes to enable an `OP_EXPIRE` opcode that would
+      make a transaction invalid for inclusion after a specified block
+      height if the transaction's script executes `OP_EXPIRE`.  This can
+      be used to make Mallory's preimage condition of an HTLC only
+      usable before Bob's refund condition becomes spendable.  This
+      prevents Mallory from being able to replace Bob's refund spend,
+      making it impossible for Mallory to execute a replacement cycle
+      attack.  `OP_EXPIRE` may also address some [transaction pinning
+      attacks][topic transaction pinning] against HTLCs.
+
+        The main downside of `OP_EXPIRE` is that it requires changes to
+        consensus to enable and changes to relay and mempool policy to
+        avoid certain problems, such as it being used to waste node
+        bandwidth.
+
+        A [reply][harding expire] to the proposal suggested a weaker way
+        to accomplish some of the same goals as `OP_EXPIRE` but without
+        any consensus or relay policy changes required.  However, Peter
+        Todd [argued][todd expire2] that it doesn't prevent the
+        replacement cycling attack.
+
+    Optech expects continued discussion about the subject and will
+    summarize any notable developments in future newsletters.
+
+- **Bitcoin UTXO set summary hash replacement:** Fabian Jahr
+  [posted][jahr hash_serialized_2] to the Bitcoin-Dev mailing list to
+  announce that a bug had been discovered in Bitcoin Core's calculation
+  of the hash of the current UTXO set.  The hash did not commit to the
+  height and coinbase information for each UTXO, information that is
+  needed for enforcing the 100-block coinbase maturity rule and
+  [BIP68][] relative timelocks.  All of that information is still in the
+  database of a node that's been synced from scratch (all current
+  Bitcoin Core nodes) and it's still used for enforcement, so this bug
+  does not affect any known released software.  However, the
+  experimental [assumeUTXO][topic assumeutxo] feature planned for
+  the next major version of Bitcoin Core will allow users to share their
+  UTXO databases with each other.  The incomplete commitment means that
+  a modified database could have the same hash as a verified database,
+  potentially opening a narrow window for an attack against assumeUTXO
+  users.
+
+  If you're aware of
+  any software that uses the `hash_serialized_2` field, please notify
+  its authors about the problem and encourage them to read Jahr's email
+  about the changes being made for the next major version of Bitcoin
+  Core to address the bug.
+
+- **Research into generic covenants with minimal Script language changes:**
+  Rusty Russell [posted][russell scripts] to Bitcoin-Dev a link to some
```

Comment:
```suggestion
  Rusty Russell [posted][russell scripts] to Bitcoin-Dev mailing list a link to some
```

Reply:
Took this suggestion with a slight addition